### PR TITLE
Resume the work

### PR DESCRIPTION
**Pull request by rockstorm101**
I'm completely new to this MIME type thing so please review this and feel free to make any changes you see fit.

I'm pushing to your repository hoping that you'll submit this addition to freedesktop.org. I'd rather have you doing this, but in case you are too busy or you prefer me doing it, let me know and I'll try.

### CODE REVIEW

File: freedesktop.org.xml.in [hunk 1]
```diff
@@ -6316,4 +6320,15 @@ command to generate the output files.
     <glob pattern="*.stl"/>
   </mime-type>
 
+<mime-type type="application/x-gcode">
+    <_comment>G-code file</_comment>
+    <sub-class-of type="text/plain"/>
+    <generic-icon name="text-x-generic"/>
+    <magic priority="10">
+      <match type="string" value="; " offset="0"/>
+      <match type="string" value="( " offset="0"/>
```
Comment (hroncok) [inline freedesktop.org.xml.in]: What exactly does this do?

Comment (rockstorm101) [reply freedesktop.org.xml.in]: The gcode and cnc files I know typically start with a comment stating the name of the file, the date it was created,... I copied this "behavior" from the C source files entry which include "//" and "/*" as a way of identifying them.

Comment (hroncok) [reply freedesktop.org.xml.in]: I meant the `( `

**Comment by hroncok** [comment]
I wanted to submit the complete set with PLY, OBJ, AMF atc. But maybe we should submit this to see what the feedback would be. Once merged and rebased I'll submit it and will keep you CC'ed.

I am new to MIME type as well. That's why I never finished this.

File: freedesktop.org.xml.in (hunk 1 above)
Comment (rockstorm101) [reply freedesktop.org.xml.in]: The parenthesis are just another way of commenting G-code. As in http://www.cerebralmeltdown.com/download/CamBamG-codeDXFandSTL/gcode/smile.nc for instance.

Comment (hroncok) [reply freedesktop.org.xml.in]: Aah. Never seen that in RepRap related GCODE.

What about adding `G` for files starting without comments? I found an old Skeinforge generated Gcode on my drive and it doesn't have any:

```
G90
G21
G28                
G92 E0
G1 E2 F1200
G1 E1 F1200
G92 E0
G1 X5.0 Z0.3 F200
G1 Y75.0 E2.0  F1000.0
G1 Y100.0 E2.3  F1000.0
G90
G21
M103
M105
M113 S1.0
M108 S12.86
G1 X60.844659 Y90.0 Z0.3 F7800.0
G1 F2040.0
G1 E1.1
G1 F7800.0
M101
G1 X61.749443 Y98.292453 Z0.3 F840.0 E1.2842446
...
```

Comment (hroncok) [reply freedesktop.org.xml.in]: Also Cura Lulzbot does this (no space):

```
;Basic settings: Layer height: 0.25 Walls: 1 Fill: 20
;Print time: 3 hours 15 minutes
;Filament used: 7.411m 60.0g
;Filament cost: None
;M190 R110 ;Uncomment to add your own bed temperature line
;M109 R240 ;Uncomment to add your own temperature line
G26                          ; clear potential 'probe fail' condition
...
```

**Comment by rockstorm101** [comment]
> I wanted to submit the complete set with PLY, OBJ, AMF atc. But maybe we should submit this to see what the feedback would be.

Actually, there's no hurry at all, this matter has been on hold for two years, it can surely wait a couple of months.

> Once merged and rebased I'll submit it and will keep you CC'ed.

Perfect. Thanks a lot.

File: freedesktop.org.xml.in (hunk 1 above)
Comment (rockstorm101) [reply freedesktop.org.xml.in]: > Aah. Never seen that in RepRap related GCODE.

Me neither, but I researched a bit to see if we could include more CNC machines and I came across that way of commenting.

I thought about adding `G` or `M` for files with no comments but there could be a lot plain-text-like files starting this way, however starting with `;` seems rather singular. Starting with a `(` might be pretty common too though.

> Also Cura Lulzbot does this (no space)

The space should be removed then.

**Comment by hroncok** [comment]
https://bugs.freedesktop.org/show_bug.cgi?id=99421

I wasn't able to add you to CC with the e-mail address you've used in the commits, because there's no such account registered in FreeDesktop's bugzilla.

**Comment by rockstorm101** [comment]
> I wasn't able to add you to CC with the e-mail address you've used in the commits, because there's no such account registered in FreeDesktop's bugzilla.

I got myself an account in case it is necessary.

**Pull request merged**